libraries/ms-drivers/src/ads1259_adc.c

```diff
@@ -0,0 +1,136 @@
+  uint8_t payload[] = { command };
+  spi_exchange(storage->spi_port, payload, 1, NULL, 0);
+}
+
+// Reads 1-byte reg value to storage->data
+static StatusCode prv_check_register(Ads1259Storage *storage, uint8_t reg_add, uint8_t reg_val) {
+  uint8_t payload[] = { (ADS1259_READ_REGISTER | reg_add) };
+  spi_exchange(storage->spi_port, payload, 1, &storage->rx_data.MSB, 1);
+  if (reg_val != storage->rx_data.MSB) return STATUS_CODE_UNINITIALIZED;
+  return STATUS_CODE_OK;
+}
+
+static StatusCode prv_configure_registers(Ads1259Storage *storage) {
+  uint8_t register_lookup[NUM_CONFIG_REGISTERS] = {
+    (ADS1259_SPI_TIMEOUT_ENABLE),
+    (ADS1259_OUT_OF_RANGE_FLAG_ENABLE | ADS1259_CHECK_SUM_ENABLE),
+    (ADS1259_CONVERSION_CONTROL_MODE_PULSE | ADS1259_DATA_RATE_SPS),
+  };
+  // reset all register values to default
+  prv_send_command(storage, ADS1259_RESET);  // Needs 8 fclk cycles before next command
+  uint8_t payload[NUM_REGISTER_WRITE_COMM] = { (ADS1259_WRITE_REGISTER | ADS1259_ADDRESS_CONFIG0),
+                                               NUM_CONFIG_REGISTERS, register_lookup[0],
+                                               register_lookup[1], register_lookup[2] };
+  // tx write-reg command and data for all three config registers
+  spi_exchange(storage->spi_port, payload, NUM_REGISTER_WRITE_COMM, NULL, 0);
+  // sanity check that data was written correctly
+  for (int reg = 0; reg < NUM_CONFIG_REGISTERS; reg++) {
+    status_ok_or_return(prv_check_register(storage, reg, register_lookup[reg]));
+  }
+  return STATUS_CODE_OK;
+}
+
+// calculate check-sum based on page 29 of datasheet
+static Ads1259StatusCode prv_checksum(Ads1259Storage *storage) {
+  uint8_t sum = (uint8_t)(storage->rx_data.LSB + storage->rx_data.MID + storage->rx_data.MSB +
+                          ADS1259_CHECKSUM_OFFSET);
+  if (storage->rx_data.CHK_SUM & CHK_SUM_FLAG_BIT) return ADS1259_STATUS_CODE_OUT_OF_RANGE;
+  if ((sum &= ~(CHK_SUM_FLAG_BIT)) != (storage->rx_data.CHK_SUM &= ~(CHK_SUM_FLAG_BIT))) {
+    return ADS1259_STATUS_CODE_CHECKSUM_FAULT;
+  }
+  return ADS1259_STATUS_CODE_OK;
+}
+
+// using the amount of noise free bits based on the SPS and VREF calculate analog voltage value
+static void prv_convert_data(Ads1259Storage *storage) {
+  double resolution = pow(2, s_num_usable_bits[ADS1259_DATA_RATE_SPS]);
+  storage->reading = (storage->conv_data.raw >> (24 - s_num_usable_bits[ADS1259_DATA_RATE_SPS])) *
+                     EXTERNAL_VREF / resolution;
+}
+
+static void prv_conversion_callback(SoftTimerId timer_id, void *context) {
+  Ads1259Storage *storage = (Ads1259Storage *)context;
+  Ads1259StatusCode code;
+  uint8_t payload[] = { ADS1259_READ_DATA_BY_OPCODE };
+  spi_exchange(storage->spi_port, payload, 1, (uint8_t *)&storage->rx_data, NUM_ADS_RX_BYTES);
+  code = prv_checksum(storage);
+  if (code) {
+    (*storage->handler)(code, NULL);
+  }
+  storage->conv_data.MSB = storage->rx_data.LSB;
```

Comment:
Why are you flipping these? the MSB should always be the MSB since you flipped the order in the structs